src/backend/marsha/core/tests/test_models_liveregistration.py

```diff
@@ -0,0 +1,87 @@
+        with self.assertRaises(ValidationError) as context:
+            LiveRegistrationFactory(email=None)
+        self.assertEqual(context.exception.messages, ["This field cannot be null."])
+
+    def test_models_liveregistration_fields_id_unique(self):
+        """Duo email/video should be unique for a given duo."""
+        video = VideoFactory()
+        video2 = VideoFactory()
+
+        # sophie@test-fun-mooc.fr register to video
+        liveregister = LiveRegistrationFactory(
+            email="sophie@test-fun-mooc.fr", video=video
+        )
+
+        # sophie@test-fun-mooc.fr can register to video2
+        LiveRegistrationFactory(email="sophie@test-fun-mooc.fr", video=video2)
+
+        # Trying to create a video with the same duo email/video should raise a
+        # database error
+        with self.assertRaises(IntegrityError):
+            with transaction.atomic():
+                LiveRegistrationFactory(email="sophie@test-fun-mooc.fr", video=video)
+
+        # Soft deleted videos should not count for unicity
+        liveregister.delete(force_policy=SOFT_DELETE_CASCADE)
+        LiveRegistrationFactory(email="sophie@test-fun-mooc.fr", video=video)
+
+    def test_models_liveregistration_fields_lti_id_unique(self):
+        """Combinaison of lti_user_id/lti_consumer_site/video should be unique."""
+        video = VideoFactory()
+        video2 = VideoFactory()
+
+        liveregister = LiveRegistrationFactory(
+            email="registered@test-fun-mooc.fr",
+            video=video,
+            lti_user_id="56255f3807599c377bf0e5bf072359fd",
+            lti_consumer_site=video.playlist.consumer_site,
+        )
+
+        # A registration with a different email for another video with the same lti info
+        # is possible
+        LiveRegistrationFactory(
+            email="issaved@test-fun-mooc.fr",
+            video=video2,
+            lti_user_id="56255f3807599c377bf0e5bf072359fd",
+            lti_consumer_site=video.playlist.consumer_site,
+        )
+
+        # Trying to create a video with the same trio lti_user_id/lti_consumer_site/video
+        # should raise a database error
+        with self.assertRaises(ValidationError) as context:
+            LiveRegistrationFactory(
+                email="sophie@test-fun-mooc.fr",
+                lti_user_id="56255f3807599c377bf0e5bf072359fd",
+                lti_consumer_site=video.playlist.consumer_site,
+                video=video,
+            )
+
+        self.assertEqual(
+            context.exception.messages,
+            [
+                (
+                    "Live registration with this LTI user identifier, "
+                    + "LTI consumer site and Video already exists."
+                )
+            ],
+        )
+
+        # Soft deleted videos should not count for unicity
+        liveregister.delete(force_policy=SOFT_DELETE_CASCADE)
+        LiveRegistrationFactory(email="sophie@test-fun-mooc.fr", video=video)
```

Comment:
You are not testing the same unicity here. You should test with `lti_user_id` and `lti_consumer_site`

Reply:
yes indeed thank you